packages/xrpl/test/integration/transactions/nftokenMint.test.ts

```diff
@@ -1,67 +1,87 @@
 import { assert } from 'chai'
-import _ from 'lodash'
-import { Client } from 'xrpl'
+import { TransactionMetadata, TxRequest } from 'xrpl'
 
+import { convertStringToHex, getNFTokenID, NFTokenMint } from '../../../src'
+import { hashSignedTx } from '../../../src/utils/hashes'
+import serverUrl from '../serverUrl'
 import {
-  convertStringToHex,
-  getNFTokenID,
-  NFTokenMint,
-  TransactionMetadata,
-} from '../../../src'
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { testTransaction } from '../utils'
 
 // how long before each test case times out
 const TIMEOUT = 20000
 
 describe('NFTokenMint', function () {
-  // TODO: Once we update our integration tests to handle NFTs, replace this client with XrplIntegrationTestContext
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
   it(
     'get NFTokenID',
     async function () {
-      const client = new Client('wss://s.altnet.rippletest.net:51233/')
-      await client.connect()
-
-      const { wallet, balance: _balance } = await client.fundWallet()
-
       const tx: NFTokenMint = {
         TransactionType: 'NFTokenMint',
-        Account: wallet.address,
+        Account: testContext.wallet.address,
         URI: convertStringToHex('https://www.google.com'),
         NFTokenTaxon: 0,
       }
-      try {
-        const response = await client.submitAndWait(tx, {
-          wallet,
-        })
-        assert.equal(response.type, 'response')
-        assert.equal(
-          (response.result.meta as TransactionMetadata).TransactionResult,
-          'tesSUCCESS',
-        )
+      const response = await testTransaction(
+        testContext.client,
+        tx,
+        testContext.wallet,
+      )
+      assert.equal(response.type, 'response')
 
-        const accountNFTs = await client.request({
-          command: 'account_nfts',
-          account: wallet.address,
-        })
+      const txRequest: TxRequest = {
+        command: 'tx',
+        transaction: hashSignedTx(response.result.tx_blob),
+      }
+      const txResponse = await testContext.client.request(txRequest)
```

Comment:
To clarify are you saying that the `tx` request is unnecessary, or that the check that the `txResponse.meta.TransactionResult` is `tesSUCCESS` is unnecessary? 

I think it's ok to omit the `tesSUCCESS` check, although it's slightly different because it's making sure the metadata is up to date (which should always be the case), but the tx request in general is needed for the rest of the checks later on.

Reply:
Both - I thought the `testTransaction` method returned a response that you could use for the rest of the checks.

Reply:
`testTransaction` returns a `SubmitResponse` which is a bit different than a `TxResponse` - notably it doesn't have the meta field, which is needed to get the `NFTokenID`.